addons/l10n_jo_edi/models/account_move.py

```diff
@@ -0,0 +1,198 @@
+        comodel_name="ir.attachment",
+        string="Jordan E-Invoice XML",
+        compute=lambda self: self._compute_linked_attachment_id(
+            "l10n_jo_edi_xml_attachment_id", "l10n_jo_edi_xml_attachment_file"
+        ),
+        depends=["l10n_jo_edi_xml_attachment_file"],
+        help="Jordan: e-invoice XML.",
+    )
+
+    @api.depends("country_code", "move_type")
+    def _compute_l10n_jo_edi_is_needed(self):
+        for move in self:
+            move.l10n_jo_edi_is_needed = (
+                move.country_code == "JO"
+                and move.move_type in ("out_invoice", "out_refund")
+            )
+
+    @api.depends("l10n_jo_edi_state")
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS 'account'
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda move: move.l10n_jo_edi_state == 'sent').show_reset_to_draft_button = False
+
+    @api.depends("l10n_jo_edi_is_needed")
+    def _compute_l10n_jo_edi_uuid(self):
+        for invoice in self:
+            if invoice.l10n_jo_edi_is_needed and not invoice.l10n_jo_edi_uuid:
+                invoice.l10n_jo_edi_uuid = uuid.uuid4()
+
+    def _l10n_jo_qr_code_src(self):
+        self.ensure_one()
+        encoded_params = url_encode({
+            'barcode_type': 'QR',
+            'value': self.l10n_jo_edi_qr,
+            'width': 200,
+            'height': 200,
+        })
+        return f'/report/barcode/?{encoded_params}'
+
+    def _is_sales_refund(self):
+        self.ensure_one()
+        return self.company_id.l10n_jo_edi_taxpayer_type == 'sales' and self.move_type == 'out_refund'
+
+    def button_draft(self):
+        # EXTENDS 'account'
+        self.write(
+            {
+                "l10n_jo_edi_error": False,
+                "l10n_jo_edi_state": False,
+            }
+        )
+        return super().button_draft()
+
+    def action_post(self):
+        # EXTENDS 'account'
+        for invoice in self.filtered('l10n_jo_edi_is_needed'):
+            invoice.l10n_jo_edi_state = 'to_send'
+        return super().action_post()
+
+    def _l10n_jo_build_jofotara_headers(self):
+        self.ensure_one()
+        return {
+            'Client-Id': self.company_id.l10n_jo_edi_client_identifier,
+            'Secret-Key': self.company_id.l10n_jo_edi_secret_key,
+            'Content-Type': 'application/json',
+        }
+
+    def _submit_to_jofotara(self, xml_invoice):
+        self.ensure_one()
+        headers = self._l10n_jo_build_jofotara_headers()
+        params = json.dumps({'invoice': xml_invoice.decode('utf-8')})
+
+        try:
+            response = requests.post(JOFOTARA_URL, data=params, headers=headers, timeout=50)
+        except requests.exceptions.Timeout:
+            return _("Request time out! Please try again.")
+        except (requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
+            return _("Invalid request: %s", e)
```

Comment:
@aboo-odoo There is a reason why this noqa is put: namely that we should not do this.  It is better for things to fail so we see the tracebacks than to fail silently.  I would not add the except Exception here.

Reply:
Yep, true my bad. There is actually a nicer way, `HTTPError` will never trigger since `RequestError` is the top level of `request` error implementation. So something like:
```suggestion
        try:
            response = requests.post(JOFOTARA_URL, data=params, headers=headers, timeout=50)
        except requests.exceptions.Timeout:
            return _("Request time out! Please try again.")
        except requests.exceptions.RequestException as e:
            return _("Invalid request: %s", e)
```

Would catch all requests errors other than a Timeout